content/blog/setting-up-hierarchical-namespaces-in-kubernetes-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,583 @@
+
+K8s namespaces created within a cluster are peers, each fully isolated from the others. K8s namespaces don't align well with organizational structures. Cluster adminitrators must define roles and create policies and constraints for each individual namespace. At scale, managing numerous namespaces can become challenging, leading to potential management issues. The administrative overhead can make managing namespaces within the cluster tedious and prone to errors.
+
+
+
+
+
+In 2020, K8s upstream introduced a K8s extension known as the [*Hierarchical Namespace Controller* (HNC)](https://github.com/kubernetes-sigs/hierarchical-namespaces#the-hierarchical-namespace-controller-hnc). HNC supports hierarchical namespaces and helps cluster adminitrators manage the security and capabilities of namespaces with less effort than the flat, peer-to-peer namespace model. Using HNC, administrators can organize namespaces according to an organizational hierarchy and allocate capabilities accordingly. 
+
+This blog post outlines the steps to set up hierarchical namespaces in K8s in HPE GreenLake for Private Cloud Enterprise. It uses a list of K8s resources, such as [*Role*](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole)/[*RoleBinding*](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding), [*ResourceQuota*](https://kubernetes.io/docs/concepts/policy/resource-quotas/) and [*Secret*](https://kubernetes.io/docs/concepts/configuration/secret/), to showcase the benefits that hierarchical namespaces bring to K8s.
+
+### Prerequisites
+
+
+
+Before starting, make sure you have the following:
+
+* A K8s cluster being provisioned, using [HPE GreenLake Terraform provider](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/) for example, in HPE GreenLake for Private Cloud Enterprise
+* The *kubectl* CLI tool, together with the kubeconfig file for accessing the K8s cluster
+* [cURL](https://curl.se/) CLI tool
+
+
+
+### Set up hierarchical namespaces
+
+K8s does not come with hierarchical namespace support by default. There are  two components, the *HNC manager* and the optional kubectl plugin *kubectl-hns*, need to install in order to support hierarchical namespaces in K8s. 
+
+#### Install the HNC manager
+
+The HNC manager can be installed within the control plane of the K8s cluster by typing the following commands:
+
+```shell
+$ HNC_VERSION=v1.1.0
+
+$ kubectl apply -f https://github.com/kubernetes-sigs/multi-tenancy/releases/download/hnc-$HNC_VERSION/hnc-manager.yaml

```

Comment:
Sorry, Denis !

Since HNC v0.9.0, it's been moved to the new repo. I installed the latest v1.1.0 in my client, however, I was still referrring to the old HNC installation README. 

The following should be the correct HNC installation steps:

$ HNC_VERSION=v1.1.0
$ HNC_VARIANT=default

$ kubectl apply -f https://github.com/kubernetes-sigs/hierarchical-namespaces/releases/download/${HNC_VERSION}/${HNC_VARIANT}.yaml